src/front-end/typescript/lib/app/types.ts

```diff
@@ -110,7 +110,13 @@ export type Route =
   | ADT<"proposalCWUView", PageProposalCWUView.RouteParams>
   | ADT<"proposalCWUExportOne", PageProposalCWUExportOne.RouteParams>
   | ADT<"proposalCWUExportAll", PageProposalCWUExportAll.RouteParams>
-  | ADT<"proposalList", PageProposalList.RouteParams>;
+  | ADT<"proposalList", PageProposalList.RouteParams>
+  | ADT<"twuMinistryGuide", null>
+  | ADT<"twuVendorGuide", null>
+  | ADT<"cwuMinistryGuide", null>
+  | ADT<"cwuVendorGuide", null>
+  | ADT<"swuMinistryGuide", null>
+  | ADT<"swuVendorGuide", null>;
```

Comment:
The solution you have works fine. My only suggestion, which is subjective, is to use a single route value constructor:

```ts
type OpportunityType = "twu" | "cwu" | "swu";
type GuideAudience = "vendor" | "ministry";

//...
  | ADT<"guide", [OpportunityType, GuideAudience]>;
```

This would reduce the boilerplate and allow you to use `switch` statements in the single `"guide"` page component to fork logic based on opportunity type and audience.

Reply:
BTW, the `OpportunityType` and `GuideAudience` types are illustrative, you may have existing types that your prefer to use.

Reply:
Also, to clarify, the URL template would be something like: `/guide/:opportunityType/:audience`